code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm

```diff
@@ -214,3 +214,79 @@
 	environment_smash = ENVIRONMENT_SMASH_STRUCTURES
 	see_in_dark = 8
 	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_INVISIBLE
+
+
+	//Legion infested mobs
+
+/obj/effect/mob_spawn/human/corpse/damaged/legioninfested/Initialize()
+	var/type = pickweight(list("Miner" = 70, "Ashwalker" = 12, "Golem" = 12, "Shadow" = 2, "YeOlde" = 2, "Operative" = 2))
+	switch(type)
+		if("Miner")
+			mob_species = pickweight(list(/datum/species/human = 70, /datum/species/lizard = 26, /datum/species/fly = 2, /datum/species/plasmaman = 2))
+			if(mob_species == /datum/species/plasmaman)
+				uniform = /obj/item/clothing/under/plasmaman
+				head = /obj/item/clothing/head/helmet/space/plasmaman
+				belt = /obj/item/weapon/tank/internals/plasmaman/belt
+			else
+				uniform = /obj/item/clothing/under/rank/miner/lavaland
+			if(mob_species != /datum/species/lizard)
+				shoes = /obj/item/clothing/shoes/workboots/mining
+			gloves = /obj/item/clothing/gloves/color/black
+			mask = /obj/item/clothing/mask/gas/explorer
+			if(prob(20))
+				suit = pickweight(list(/obj/item/clothing/suit/hooded/explorer = 18, /obj/item/clothing/suit/hooded/cloak/goliath = 2))
+			if (belt == -1 && prob(4))
```

Comment:
See corpse.dm. The variables are initialized to -1 instead of null. Setting them allows you to override a specified outfit datum's equipment. -1 means don't change the datum's equipment for that slot, null means remove it, while a type path means it should be that path instead of what the outfit normally has.